woodwork/column_accessor.py

```diff
@@ -192,15 +197,14 @@ def wrapper(*args, **kwargs):
 
                 # Try to initialize Woodwork with the existing Schema
                 if _is_series(result):
-                    valid_dtype = _get_valid_dtype(result, self._schema['logical_type'])
-                    if result.dtype == valid_dtype:
+                    valid_dtype = _get_valid_underlying_dtype(result, self._schema['logical_type'])
+                    if str(result.dtype) == valid_dtype:
```

Comment:
We need to make sure we only compare dtypes' strings, because if a column's dtype is `int64`, it'll return True if we check for equality with both the strings `'int64'` (expected) and `'Int64'` (which seems less expected to me--haven't been able to figure out if this is a bug)